### PR TITLE
feat(pipeline/execution): Adds list cmd for executions.

### PR DESCRIPTION
**Pull request by jtk54**

### CODE REVIEW

File: cmd/pipeline/execution/list.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+		statuses = append(statuses, "CANCELED")
+	}
+	if len(statuses) > 0 {
+		query["statuses"] = strings.Join(statuses, ",")
+	}
+
+	if options.limit > 0 {
+		query["limit"] = options.limit
+	}
+	//fmt.Println(query)
```
Comment (ezimanyi) [inline cmd/pipeline/execution/list.go]: Unintentionally left here?

Comment (jtk54) [reply cmd/pipeline/execution/list.go]: Yep, removed.

**Comment by jtk54** [comment]
We could add a new endpoint that handles the `succeeded` etc flags in the longer term but that is a bit of scope expansion. For now I'm going to drop a TODO to do the refactor/simplification of that endpoint.

**Comment by jtk54** [comment]
Closes https://github.com/spinnaker/spinnaker/issues/3999.

**Pull request merged**